src/aerialmap_display.cpp

```diff
@@ -490,101 +498,93 @@ void AerialMapDisplay::transformAerialMap()
-    setStatus(StatusProperty::Error, "Transform", QString::fromStdString(errMsg));
-    return;
-  }
-
-  // Set the pseudo fixed frame to map so that we can align the tiles to north just by setting identity as the
-  // orientation
-  context_->getFrameManager()->setFixedFrame(mapFrame);
-
-  // Get the latest transform between the robot frame and fixed frame from the FrameManager
-  if (!context_->getFrameManager()->getTransform(frame, ros::Time(), position, orientation))
+  if (!context_->getFrameManager()->transform(ref_fix_.header.frame_id, ros::Time(), origin, position, orientation))
   {
     // display error
     std::string error;
-    if (context_->getFrameManager()->transformHasProblems(frame, ros::Time(), error))
+    if (context_->getFrameManager()->transformHasProblems(ref_fix_.header.frame_id, ros::Time(), error))
     {
       setStatus(StatusProperty::Error, "Transform", QString::fromStdString(error));
     }
     else
     {
       setStatus(StatusProperty::Error, "Transform",
-                "Could not transform from [" + QString::fromStdString(frame) + "] to Fixed Frame [" + fixed_frame_ +
-                    "] for an unknown reason");
+                "Could not transform from [" + QString::fromStdString(ref_fix_.header.frame_id) + "] to Fixed Frame [" +
+                    fixed_frame_ + "] for an unknown reason");
     }
     return;
   }
+  setStatus(StatusProperty::Ok, "Transform", "Transform OK");
+  scene_node_->setPosition(position);
+  scene_node_->setOrientation(orientation);
+}
 
-  if (position.isNaN())
+void AerialMapDisplay::computeTransformations()
+{
+  std::string errMsg;
+  if (context_->getFrameManager()->frameHasProblems(context_->getFrameManager()->getFixedFrame(), ros::Time{},
+                                                    errMsg) ||
+      context_->getFrameManager()->transformHasProblems(context_->getFrameManager()->getFixedFrame(), ros::Time{},
+                                                        errMsg) ||
+      context_->getFrameManager()->frameHasProblems(frame_property_->getFrameStd(), ros::Time{}, errMsg) ||
+      context_->getFrameManager()->transformHasProblems(frame_property_->getFrameStd(), ros::Time{}, errMsg) ||
+      context_->getFrameManager()->frameHasProblems(ref_fix_.header.frame_id, ros::Time{}, errMsg) ||
+      context_->getFrameManager()->transformHasProblems(ref_fix_.header.frame_id, ros::Time{}, errMsg))
   {
-    // This can occur if an invalid TF is published.
-    // Show an error and don't apply anything, so OGRE does not throw an assertion.
-    setStatus(StatusProperty::Error, "Transform", "Received invalid transform");
+    setStatus(StatusProperty::Error, "Transform", QString::fromStdString(errMsg));
+    setStatus(StatusProperty::Error, "README",
+              QString::fromStdString("Tiles might be visible, but in the wrong place!"));
+    // todo: replace this readme by hiding the tiles in a way
```

Comment:
This todo has to be fixed.

Reply:
True, but it's not an issue I introduced, it always existed, and I do want to contribute, yet not fix everything ;)

see https://github.com/gareth-cross/rviz_satellite/issues/51